### PR TITLE
[parents_migration] Add and backfill Google Drive "Shared with me" folders

### LINKED ISSUE #9502: Google Drive: create folders for "Shared With Me"
Similarly to [#9457](https://github.com/dust-tt/dust/issues/9457), every google-drive connector should have a "Shared With Me" folder node, considered as top-level parent for files that are shared with the user without rendition of a hierarchy.

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Close #9502
- AFAICT we never set `GOOGLE_DRIVE_SHARED_WITH_ME_VIRTUAL_ID` as a `parentInternalId` but it still works because we return the correct nodes when querying getPermissions with `parent=GOOGLE_DRIVE_SHARED_WITH_ME_VIRTUAL_ID`.

## Risk

- Lower than all the other parents migrations, easily rollbackable.

## Deploy Plan

- Deploy connectors.
- Run migration.

### CODE REVIEW

**Review by philipperolet — COMMENTED** [review]
Looks good, left a com

File: connectors/src/resources/connector/google_drive.ts [hunk 1]
```diff
@@ -71,6 +72,10 @@ export class GoogleDriveConnectorStrategy
       },
       transaction,
     });
+    await deleteDataSourceFolder({
```
[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply connectors/src/resources/connector/google_drive.ts]: didn't mean to put it here mb, moved it to the clean method of the ConnectorManager

[the remark this answers is not in the document]
Comment (philipperolet) [reply connectors/src/resources/connector/google_drive.ts]: not sure it goes there either; although maybe it should
i'd say it goes where we also delete documents and tables
(note: logged the task [here](https://github.com/dust-tt/dust/issues/9512))

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply connectors/src/resources/connector/google_drive.ts]: I would put all these deletes here too I think, can't really think of another place

File: connectors/src/connectors/google_drive/index.ts [hunk 2]
```diff
@@ -129,6 +131,17 @@ export class GoogleDriveConnectorManager extends BaseConnectorManager<null> {
       googleDriveConfigurationBlob
     );
 
+    // Upserts to data_sources_folders (core) a top-level folder "Shared with me".
+    const folderId = getInternalId(GOOGLE_DRIVE_SHARED_WITH_ME_VIRTUAL_ID);
+    await upsertDataSourceFolder({
```
Comment (philipperolet) [inline connectors/src/connectors/google_drive/index.ts:136]: :cough: :cough: activity :wink:

File: connectors/migrations/20241218_backfill_gdrive_shared_with_me.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { upsertDataSourceFolder } from "@connectors/lib/data_sources";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+makeScript({}, async ({ execute }, logger) => {
+  const connectors = await ConnectorResource.listByType("google_drive", {});
+
+  await concurrentExecutor(
+    connectors,
+    async (connector) => {
+      const folderId = getInternalId(GOOGLE_DRIVE_SHARED_WITH_ME_VIRTUAL_ID);
```
Comment (philipperolet) [inline connectors/migrations/20241218_backfill_gdrive_shared_with_me.ts:16]: just checking: you told me documents already have this folderId in their parents right? (it was just not pointing to any real folder until this PR)

**Review by philipperolet — COMMENTED** [review]
Looks good :+1: Left 3 coms

File: connectors/migrations/20241218_backfill_gdrive_shared_with_me.ts (hunk 3 above)
Comment (aubin-tchoi) [reply connectors/migrations/20241218_backfill_gdrive_shared_with_me.ts:16]: actually we never pass this as a parentId, I kept this PR iso-behavior with main but I can't figure out how it used to work. I'll check again and fix it if needed

**Review by philipperolet — APPROVED** [review]

**Pull request merged**